cmd/serverless-init/metric/metric.go

```diff
@@ -12,20 +12,29 @@ import (
 
 	"github.com/DataDog/datadog-agent/pkg/aggregator"
 	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // AddColdStartMetric adds the coldstart metric to the demultiplexer
 //
 //nolint:revive // TODO(SERV) Fix revive linter
 func AddColdStartMetric(metricPrefix string, tags []string, _ time.Time, demux aggregator.Demultiplexer) {
-	add(fmt.Sprintf("%v.enhanced.cold_start", metricPrefix), tags, time.Now(), demux)
+	if demux != nil {
+		add(fmt.Sprintf("%v.enhanced.cold_start", metricPrefix), tags, time.Now(), demux)
+	} else {
+		log.Debugf("Unable to add cold start metric. The metric agent is not running")
+	}
 }
 
 // AddShutdownMetric adds the shutdown metric to the demultiplexer
 //
 //nolint:revive // TODO(SERV) Fix revive linter
 func AddShutdownMetric(metricPrefix string, tags []string, _ time.Time, demux aggregator.Demultiplexer) {
-	add(fmt.Sprintf("%v.enhanced.shutdown", metricPrefix), tags, time.Now(), demux)
+	if demux != nil {
+		add(fmt.Sprintf("%v.enhanced.shutdown", metricPrefix), tags, time.Now(), demux)
+	} else {
+		log.Debugf("Unable to add shudtdown metric. The metric agent is not running")
+	}
```

Comment:
I would actually recommend moving this nil check into the `add` function below. It allows you to consolidate the checks into a single location and also means that you'll never have to worry about it again in the future should you wish to create a new metric someday.

Additionally, this is a nit, but in go it's common to see nil checks at the top of a function in this style:

```golang
func add(name string, tags []string, timestamp time.Time, demux aggregator.Demultiplexer) {
	if demux == nil {
		log.Debugf("Unable to add metric %s, the metric agent is not running", name)
		return
	}
	... do the thing ...
}
```

Reply:
Good call